CorinneTest/PersonDataSource.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CorinneTest
+// singleton for person list
+{
+    public class PersonDataSource
+    {
+
+        #region Singleton
+        private static PersonDataSource instance = null;
+
+        public static PersonDataSource Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    instance = new PersonDataSource();
+                }
+                return instance;
+            }
+        }
+
+        private PersonDataSource()
+        {
+            this.People = new List<Person>();
+        }
+        #endregion
+
+        #region Variables
+        public List<Person> People { get; set; }
```

Comment:
Probably want to use List<IPerson>

Reply:
Either List without the <> (at which point the T-Type would be the base "object" class ala Java.Lang.Object) or make it List<IPerson> (assuming you implement the interface)

Reply:
using the non-generic one (`List` with no `<>`) is very undesirable don't you think @gregoryfein ? You'd need to cast each object as you go or use reflection to determine types and you lose all of your type safety.

Reply:
@nnieslan agreed.  but that was going to be part of my exercise about providing the right generic and how generic do you want to be ;)

Reply:
@nnieslan what do you mean use List? like not have a List of type Person?